plugin-server/src/main/ingestion-queues/session-recording/blob-ingester/session-offset-high-water-mark.ts

```diff
@@ -0,0 +1,164 @@
+                })
+                this.topicPartitionWaterMarks[`${tp.topic}-${tp.partition}`][sessionId] = offset
+            })
+        } catch (error) {
+            status.error('🧨', 'WrittenOffsetCache failed to add high-water mark for partition', {
+                error: error.message,
+                key,
+                ...tp,
+                sessionId,
+                offset,
+            })
+            captureException(error, {
+                extra: {
+                    key,
+                    offset,
+                },
+                tags: {
+                    ...tp,
+                    sessionId,
+                },
+            })
+        }
+    }
+
+    public async getAll(tp: TopicPartition): Promise<Record<string, number> | null> {
+        const key = offsetHighWaterMarkKey(this.keyPrefix, tp)
+        try {
+            return await this.run(`read all offset high-water mark for ${key} `, async (client) => {
+                const redisValue = await client.zrange(key, 0, -1, 'WITHSCORES')
+                // redisValue is an array of [sessionId, offset, sessionId, offset, ...]
+                // we want to convert it to an object of { sessionId: offset, sessionId: offset, ... }
+                const highWaterMarks = redisValue.reduce(
+                    (acc: Record<string, number>, value: string, index: number) => {
+                        if (index % 2 === 0) {
+                            acc[value] = parseInt(redisValue[index + 1])
+                        }
+                        return acc
+                    },
+                    {}
+                )
+                if (highWaterMarks) {
+                    this.topicPartitionWaterMarks[`${tp.topic}-${tp.partition}`] = highWaterMarks
+                }
+                return highWaterMarks
+            })
+        } catch (error) {
+            status.error('🧨', 'WrittenOffsetCache failed to read high-water marks for partition', {
+                error: error.message,
+                key,
+                ...tp,
+            })
+            captureException(error, {
+                extra: {
+                    key,
+                },
+                tags: {
+                    ...tp,
+                },
+            })
+            return null
+        }
+    }
+
+    public async onCommit(tp: TopicPartition, offset: number): Promise<Record<string, number> | null> {
+        const key = offsetHighWaterMarkKey(this.keyPrefix, tp)
+        try {
+            return await this.run(`commit all below offset high-water mark for ${key} `, async (client) => {
+                const numberRemoved = await client.zremrangebyscore(key, '-Inf', offset)
+                status.info('📝', 'WrittenOffsetCache committed all below high-water mark for partition', {
+                    numberRemoved,
+                    ...tp,
+                    offset,
+                })
+                return await this.getAll(tp)
```

Comment:
I assume this is here to refresh the list? 
Why don' we just do this in memory? Would be faster and have the same result